Issue 1807

richardschneider opened the pull request:
i18n of language restart dialog fixes #1807

![issue-1807-a](https://cloud.githubusercontent.com/assets/631061/19543468/1c678d6e-96d4-11e6-8a83-12a047c387a1.JPG)
![issue-1807-b](https://cloud.githubusercontent.com/assets/631061/19543467/1c65a21a-96d4-11e6-96d9-b1da1be70743.JPG)

reedrosenbluth committed:
display shortened website url

Commit:
added enter key action

richardschneider committed:
chore(.jshintrc): configure jshint to pass most of the code

richardschneider committed:
chore(editor): configuration settings for an editor

richardschneider committed:
chore(editor): configuration settings for an editor

richardschneider committed:
fix(settings): i18n of the restart buttons

richardschneider committed:
fix(settings): i18n of the restart buttons

richardschneider committed:
fix(settings): l10n of window title

'Settings' does not exist in the localisation files, but 'nav.settings' does.

richardschneider committed:
fix(settings): l10n of window title

'Settings' does not exist in the localisation files, but 'nav.settings' does.

rmisio commented on .jshintrc:
```diff
@@ -1,10 +1,19 @@
 
   "curly": true,
   "latedef": true,
-  "quotmark": true,
```
Generally, our expectation is that code is modified to adhere to the linting rules rather than the rules adjusted to satisfy offending code.

What are your reasons for these rule changes? The one that particularly concerns me is setting 'quotmark' to false.

richardschneider replied:
It appears to me that  you guys have replaced `jshint` with `eslint`.  My editor of choice `brackets`  uses `jshint`.  I made the above changes to limit the number warnings produced by the editor.

I thought it was better to change the config, which you do not use, than making lots of changes to the code that has nothing to do with the bug I was trying to fix.

jjeffryes replied:
We don't use jshint any more, and will be removing it from the repo, thank you for bringing it to our attention. 

I noticed there were extensions available to use eslint with brackets, it would be best to use one of those.

This file shouldn't be included with your PR. If you want to submit a deleted jshint file with the PR, that would be fine.

rmisio commented on .jshintrc:
```diff
@@ -1,10 +1,19 @@
 {
   "node": true,
+	"browser": true,
```
Could you please fix the indentation here?

richardschneider replied:
Sorry about tabs, will fix and resubmit.

rmisio commented on .jshintrc:
```diff
@@ -1,10 +1,19 @@
   "unused": true,
-  "trailing": true
+  "trailing": true,
+	"expr": true,
```
Could you also fix the indentation here as well?

richardschneider committed:
chore(.jshintrc): fix indentation

richardschneider commented:
Both `eslint` and `jshint` are generating warnings for many files.  Should I raise another issue and start resolving them by fixing the code?

I've always believed that unless all warnings are fixed in the code base, contributors will be lazy.

jjeffryes commented:
We added eslint and set rules for it after a lot of the code base already existed, so while fixing warnings is appreciated, it's not a requirement.

Eslint should always be used instead of jshint.

rmisio committed:
when using the bundled app randomizing the ports the local server is launched with to minimized Port In Use errors

rmisio committed:
added extra check when setting remote ports

jjeffryes committed:
Merge pull request #1806 from ameliagoodman/enterKeyConnection

added enter key action

jjeffryes commented on .editorconfig:
```diff
@@ -0,0 +1,21 @@
+
+[*]
+
+# Change these settings to your own preference
```
If we add a editorconfig file, it should match our eslint settings (it looks like this one does). Please change the comments to indicate none of the settings should be changed, otherwise contributors may switch them to something that is flagged as incorrect by eslint.

jjeffryes reviewed (changes requested):
Thanks for the PR. If you can make the minor changes requested, we'll merge it in.

jjeffryes committed:
Updates to the latest version of electron, along with changes to require statements to match the new API.
- deletes the unused jshint files
- clean up eslint warnings
- update eslint case indent and standardize indents in cases

jjeffryes committed:
Merge pull request #1810 from rmisio/randomizePorts

when using the bundled app randomizing the ports the local server is …

rmisio committed:
Merge pull request #1811 from OpenBazaar/UpdateElectron

Update Electron

jjeffryes committed:
Merge pull request #1803 from reedrosenbluth/cosmeticLinkName

display shortened website url

richardschneider committed:
chore(editor): indent style and spacing should not be changed

richardschneider committed:
Revert "chore(editor): indent style and spacing should not be changed"

This reverts commit 939ae91eda48c81a1c6712875efa9798f55b7847.

richardschneider commented:
I've rebased to the master

richardschneider commented:
I had to rebase to master because of conflicts.  Hopefully, everything is good to go.

jjeffryes commented:
It looks like something went funny with the rebase, it's trying to re-apply changes that have already been made. I'll take a look tomorrow and see if I can get it synced properly.
